shacl12-node-expr/index.html

```diff
@@ -2647,6 +2647,142 @@ <h3>sh:nodeByExpression</h3>
 			
 		</section>
 
+		<section id="dynamic-shacl">
+			<h2>Dynamic SHACL</h2>
+			<p>
+				This section defines <dfn>Dynamic SHACL</dfn> as a dialect of SHACL that some implementations MAY support.
+			</p>
+			<p>
+				In Dynamic SHACL any <a>parameter</a> of a <a>constraint</a> can be a <a>node expression</a>,
+				excluding those that do <em>not</em> allow <a>blank nodes</a> (such as <code>sh:node</code>)
+				but including those that take <a>SHACL lists</a> as values (such as <code>sh:class</code>, <code>sh:datatype</code> and <code>sh:in</code>).
+				During <a>validation</a>, such <a>node expressions</a> are evaluated in the <a>data graph</a>,
+				using the current <a>focus node</a>.
+				The resulting <a>nodes</a> will be used as <a>parameters</a> for the <a>constraint</a>.
+			</p>
+			<section class="informative">
+				<h3>Example: Dynamic Minimum Age of Presidents</h3>
+				<p>
+					As a use case of <a>Dynamic SHACL</a>, assume we want to express that the legal minimum age
+					of a president is 18 unless the country is USA, where it is 35.
+				</p>
+				<aside class="example" title="Example of Dynamic SHACL using a node expression at sh:minInclusive">
+					<div class="shapes-graph">
+						<div class="turtle">
+ex:PresidentShape
+    a sh:NodeShape ;
+	sh:targetClass ex:President ;
+    sh:property ex:PresidentShape-age ;
+.
+ex:PresidentShape-age
+    a sh:PropertyShape ;
+    sh:path ex:age ;
+    <b>sh:minInclusive [
+        shnex:if [
+            sparql:eq (
+                [ shnex:path ex:country ]
+                ex:USA
+            )
+        ]
+        shnex:then 35 ;
+        shnex:else 18 ;
```

Comment:
```suggestion
ex:PresidentShape
    a               sh:NodeShape ;
	sh:targetClass  ex:President ;
    sh:property     ex:PresidentShape-age ;
.
ex:PresidentShape-age
    a               sh:PropertyShape ;
    sh:path         ex:age ;
    <b>sh:minInclusive [
        shnex:if [
            sparql:eq (
                [ shnex:path ex:country ]
                ex:USA
            )
        ]
        shnex:then 35 ;
        shnex:else 18 ;
```

Reply:
I don't like that to be honest. Spacing makes sense for comments but if we do this here, we would need to change everything and the choice where to place the spaces is rather arbitrary.

Reply:
Regarding the specifics of this change request, I feel that it falls into the [Priority of Constituencies](https://www.w3.org/TR/design-principles/#priority-of-constituencies), as it makes comprehension (substantially, I think) easier for the consuming reader at some (small, I think) cost to the Authors/Editors of the documents we're producing. You may disagree with this or any other specific suggestion. To my mind, each such controversial suggestion should then become a question for the WG writ large, whose decisions on such questions become implementation instructions for the Editors.

Reply:
followup in https://github.com/w3c/data-shapes/issues/528